web/src/features/public-api/types/scores.ts

```diff
@@ -0,0 +1,221 @@
+import { isPresent } from "@/src/features/manual-scoring/lib/helpers";
+import {
+  paginationZod,
+  paginationMetaResponseZod,
+  NonEmptyString,
+  ScoreDataType,
+  stringDate,
+  ScoreSource,
+} from "@langfuse/shared";
+import { z } from "zod";
+
+/**
+ * Objects
+ */
+
+const ConfigCategory = z.object({
+  label: z.string().min(1),
+  value: z.number(),
+});
+
+const operators = ["<", ">", "<=", ">=", "!=", "="] as const;
+
+const NumericData = z.object({
+  value: z.number(),
+  stringValue: z.undefined().nullish(),
+  dataType: z.literal(ScoreDataType.NUMERIC),
+});
+
+const CategoricalData = z.object({
+  value: z.number().optional().nullish(),
+  stringValue: z.string(),
+  dataType: z.literal(ScoreDataType.CATEGORICAL),
+});
+
+const BooleanData = z.object({
+  value: z.number(),
+  stringValue: z.string(),
+  dataType: z.literal(ScoreDataType.BOOLEAN),
+});
+
+const ScoreBase = z.object({
+  id: z.string(),
+  timestamp: z.union([stringDate, z.date()]),
+  projectId: z.string(),
+  name: z.string(),
+  source: z.nativeEnum(ScoreSource),
+  authorUserId: z.string().nullish(),
+  comment: z.string().nullish(),
+  traceId: z.string(),
+  observationId: z.string().nullish(),
+  configId: z.string().nullish(),
+  createdAt: z.union([stringDate, z.date()]),
+  updatedAt: z.union([stringDate, z.date()]),
+});
+
+const Score = z.discriminatedUnion("dataType", [
+  ScoreBase.merge(NumericData),
+  ScoreBase.merge(CategoricalData),
+  ScoreBase.merge(BooleanData),
+]);
+
+const BaseScoreBody = z.object({
+  id: z.string().nullish(),
+  name: NonEmptyString,
+  traceId: z.string(),
+  observationId: z.string().nullish(),
+  comment: z.string().nullish(),
+});
+
+const GetAllScoresBase = z.object({
+  id: z.string(),
+  timestamp: z.union([stringDate, z.date()]),
+  name: z.string(),
+  source: z.nativeEnum(ScoreSource),
+  comment: z.string().nullish(),
+  traceId: z.string(),
+  observationId: z.string().nullish(),
+  trace: z.object({
+    userId: z.string(),
+  }),
+  configId: z.string().nullish(),
+});
+
+export const GetAllScores = z.discriminatedUnion("dataType", [
```

Comment:
is this the response object for scores while the ones above are the request objects?